src/agent.js

```diff
@@ -1,50 +1,112 @@
 import User from './User'
 import moment from 'moment';
+import TripsRepo from './tripsRepo';
 class Agent extends User {
-  constructor(agentData) {
+  constructor(agentData, alltrips, allDestinations, allTravelers) {
     super(agentData)
-    // this.id = data.id;
-    // this.name = data.name;
+    this.agentId = agentData.agentId;
+    this.name = agentData.name;
     this.userName = agentData.userName
     this.pwd = agentData.pwd;
+    this.today = moment().format('YYYY-MM-DD');
+    this.allTrips = alltrips.tripsData.trips
+    this.allDestinations = allDestinations.destinationsData.destinations
+    this.allTravelers = allTravelers.userTripHistory.travelers
   }
-
-  //get userpending trips
+  
   getAllPendingTrips(){
-
+    const allPendings = this.allTrips.filter(trip => trip.status === 'pending');
+    return allPendings
   }
 
-  //I should be able to see and approve / deny trip requests
-  //after getting the information of all users by data
-  resolvePendingTrips(){
-
+  resolvePendingTrips(tripID, decision){
+    const allPendings = this.getAllPendingTrips();
+    const pendingTrip = allPendings.find(trip => trip.id === tripID);
+    const modifiedTrip = new Array(pendingTrip).reduce((aprovedTrip, entry) => {
+      aprovedTrip.id = entry.id;
+      aprovedTrip.status = decision;
+      aprovedTrip.suggestedActivities = []//only status or sugg activities is required
+      return aprovedTrip
+    }, {});
```

Comment:
Is this creating an array for a single trip to modify a trip's data? If it's only ever modifying one trip there may be a more straightforward approach. Not a huge issue, and its a common pattern.